ovn_k8s/common/config.py

```diff
@@ -13,48 +13,74 @@
 # limitations under the License.
 
 import ast
-import ConfigParser
 import os
 import sys
 
-import ovn_k8s
 from ovn_k8s.common.util import ovs_vsctl
 from ovn_k8s.common.util import ovn_nbctl
 from ovn_k8s.common import variables
 
+if sys.version_info < (3, 0):
+    # python 2
+    import ConfigParser
+else:
+    # python 3
+    import configparser as ConfigParser
 
 raw_config_parser = ConfigParser.RawConfigParser()
-config_path = "/etc/openvswitch"
-config_file = '%s/ovn_k8s.conf' % config_path
+config_filename = 'ovn_k8s.conf'
+if sys.platform != 'win32':
+    config_path = "/etc/openvswitch"
+else:
+    config_path = "C:\\etc"
+config_file = '%s/%s' % (config_path, config_filename)
+# Convert the path to use platform specific path delimitators
+config_file = os.path.abspath(config_file)
 config_read = raw_config_parser.read(config_file)
 if len(config_read) != 1:
-    # raw_config_parser.read returns the list of filenames which were
-    # successfully parsed.
-    # If we reach here it means that it could not read the config file from
-    # "/etc/openvswitch".  Search in default prefix '/usr/local', this is
-    # where pip is installing the config file
-    config_file = '/usr/local%s' % config_file
-    config_read = raw_config_parser.read(config_file)
-    if len(config_read) != 1:
-        # If we reach here it means the config could not be found or read
-        # from the previous locations.  This could happen if we try to run
-        # it from a virtual env.  In this case, we try to read the config
-        # from the current repository
-        absolute_path = os.path.dirname(os.path.abspath(ovn_k8s.__file__))
-        absolute_path = absolute_path[:absolute_path.rfind("/")]
-        config_file = "%s/etc/ovn_k8s.conf" % absolute_path
-        config_read = raw_config_parser.read(config_file)
-        if len(config_read) != 1:
-            # Config could not be found or read
-            sys.exit("Error when reading config file: %s" % config_file)
+    print("Could not find the config_file in '%s', using default "
+          "values" % config_file)
```

Comment:
The above is a print statement. Since config.py is also used by daemons, this does not look clean as in daemons we use vlog to log. This makes it a little tricky as we do not initialize vlog when we run binaries (we only do it for daemons).  You can probably skip the print statement altogether and simply use default values.

Also, I think we can move all the above gobal code into a function. The first time a utility makes a call to get a config value, we can see whether we have already built a cache and if not, attempt to read from the file or the default section to build the cache.